draft-ietf-mls-extensions.md

```diff
@@ -483,6 +483,43 @@ The `media_type` MAY be zero length, in which case, the media type of the
 `application_content` is interpreted as the first MediaType specified in
 `required_media_types`.
 
+## Last resort KeyPackages
+
+Type: KeyPackage extension
+
+### Description
+
+Section 10 of {{!RFC9420}} details that are required to pre-publish KeyPackages
+s.t. other clients can add them to groups asynchronously. The specification
+states that KeyPackages generally SHOULD NOT be re-used with the exception of
+"last resort" KeyPackages. Section 16.8 of {{!RFC9420}} further describes
+last-resort KeyPackages as KeyPackages that are meant to be used if all other
+KeyPackages have been exhausted. The specification, however, leaves it to the
+application to distinguish between regular and last-resort KeyPackages.
+
+The last_resort_key_package KeyPackage extension defined in this document fills
+this gap and allows clients to specifically mark KeyPackages as KeyPackages of
+last resort that MAY be used more than once in scenarios where all other
+KeyPackages have already been used.
+
+The extension allows clients that pre-publish KeyPackages to signal to the
+Delivery Service which KeyPackage(s) are meant to be used as last resort
+KeyPackages. Such KeyPackages could, for example, support specific ciphersuites
+or contain additional extensions.
```

Comment:
It's not entirely clear to me why last-resort KPs should be different from other KPs, i.e. the example is not really motivated.